### DISCUSSION TITLE
Panache Kotlin JPA Entities defined as class constructors vs. members?

### DISCUSSION THREAD

**Discussion by kenyee** [quarkusio/quarkus#24612] [Q&A] [answered]
In https://www.baeldung.com/kotlin/jpa, the recommended way of defining a class is:
```
@Entity
class Project(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long
)

But in the Panache Kotlin documentation(https://quarkus.io/guides/hibernate-orm-panache-kotlin), it's defined like this:
```
@Entity
class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}
```

Note the () vs. {} usage.
Which is the correct way of doing this?

Reason the Panache Kotlin one seems wrong is an identity column is a primary key and it should always be non-null... and when I try reading this back from the DB, it's coming in as null even though the DB is incrementing automatically according to the SQL log output from hibernate:
```
create table Project (
    id bigint generated by default as identity,
...
```

[a comment by an automated account is not shown]

**Reply by evanchooly** [reply]
Either is "correct" so far as that goes.  The first just defines a constructor that takes the ID value.  the other just defines a field and relies on a default constructor.  But both end up with ID fields on the class it just a question of how the instanced are created.
